lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.cpp

```diff
@@ -0,0 +1,122 @@
+#include "lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.h"
+
+#include "lib/Dialect/ModArith/IR/ModArithAttributes.h"
+#include "lib/Dialect/ModArith/IR/ModArithOps.h"
+#include "lib/Dialect/ModArith/IR/ModArithTypes.h"
+#include "lib/Utils/ConversionUtils/ConversionUtils.h"
+#include "llvm/include/llvm/Support/Debug.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinAttributes.h"     // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"          // from @llvm-project
+#include "mlir/include/mlir/IR/TypeUtilities.h"         // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"             // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+#define DEBUG_TYPE "arith-to-mod-arith"
+
+namespace mlir {
+namespace heir {
+namespace arith {
+
+#define GEN_PASS_DEF_ARITHTOMODARITH
+#include "lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.h.inc"
+
+static mod_arith::ModArithType convertArithType(Type type) {
+  auto modulusBitSize = type.getIntOrFloatBitWidth();
+  auto modulus = (int)(1 << (modulusBitSize - 1)) - 1;
```

Comment:
I suspect this will overflow in general, and we may want to use APInt to be more precise. But I'm fine merging this as-is for now.

Reply:
I think the very last `-1` should be removed. With -1 it's the max value, and without -1 it's the modulus, no?

We may need to use a 64-bit storage type to store the modulus in the IR, but that should be OK...

Reply:
The -1 was indeed to keep everuthong stored in a i32 number. 
Changed it, now it stored in 64b